src/data-access/repositories/mongo/mongo-ad-lib-piece-repository.ts

```diff
@@ -51,11 +54,23 @@ export class MongoAdLibPieceRepository extends BaseMongoRepository<MongoAdLibPie
       data: {
         name: adLibPiece.name,
         rank: adLibPiece._rank,
-        expectedDuration: adLibPiece.expectedDuration ?? undefined,
+        expectedDuration: this.getExpectedDuration(adLibPiece),
         sourceLayerId: adLibPiece.sourceLayerId,
         lifespan: adLibPiece.lifespan ?? undefined,
         content: adLibPiece.content
       },
     }
   }
+
+  private getExpectedDuration(adLibPiece: MongoAdLibPiece): number | undefined {
+    if (adLibPiece.expectedDuration) {
+      return adLibPiece.expectedDuration
+    }
+
+    if (adLibPiece.enable?.duration) {
+      return adLibPiece.enable.duration
+    }
+
+    return undefined
```

Comment:
Could this be a more simplified version? 😄 

```suggestion
    return adLibPiece.expectedDuration ?? adLibPiece.enable?.duration ?? undefined
```

Reply:
If only it was undefined values we checked against, then even more simplified version would be:
```typescript
return adLibPiece.expectedDuration ?? adLibPiece.enable?.duration
```
But I think that the reason it is done as if-statements is that we also want to check for `0`.
So if this should be simplified, then one could write:
```typescript
return adLibPiece.expectedDuration || adLibPiece.enable?.duration || undefined
```

Reply:
"simplified" is quite relative.
Yes, you make the code "simpler" by having fewer lines, but you make the individual lines more complex and harder to understand, because so much more is happening on a given line.

Personally, I don't think everything should be a one-liner. Especially when the one-liner can return more-than-two outcomes (which is the case here).
The code also becomes more difficult to debug using a debugger, because a single step in the debugger steps over so many things (greatly reducing one of the strongest features of the debugger).

But this all comes down to where the line between verbosity and "as few lines as possible" is drawn. For me, that's useally if the the single line returns more-than-two outcomes.

Sidenote: Verbosity is not a bad thing :)

Reply:
From the conversation, the issue I see is a misunderstanding of what is being checked. Whether that be the conditional not being clear enough (e.g. `adLibPiece.expectedDuration` is checked for both `null | undefined` and and if it is greater than `0`) or if it is an issue in how we model the duration values (e.g. `0` could be used instead of `null` or `undefined`) or both.